test/IntegrationTests/TicketService/test_delete_custom_field.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+from http import HTTPStatus
+import requests
+import logging
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s'
+)
+logger = logging.getLogger(__name__)
+
+class TestDeleteCustomFieldAPI(unittest.TestCase):
+    BASE_URL = "http://localhost:8000"
+
+    def setUp(self):
+        """Setup function to initialize valid customer GUID and custom field."""
+        logger.info("=== Initializing test setup ===")
+
+        # Assuming an endpoint `/addcustomer` to create a new customer
+        customer_url = f"{self.BASE_URL}/addcustomer"
+        response = requests.post(customer_url)
+        self.assertEqual(response.status_code, HTTPStatus.OK, "Failed to create customer")
+        self.valid_customer_guid = response.json().get("customer_guid")
+        logger.info(f"Valid customer_guid initialized: {self.valid_customer_guid}")
+
+        # Adding custom fields for the valid customer
+        custom_field_url = f"{self.BASE_URL}/custom_fields"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "field_name": "test_field",
+            "field_type": "VARCHAR(255)",
+            "required": True
+        }
+        response = requests.post(custom_field_url, json=data)
+        self.assertEqual(response.status_code, HTTPStatus.CREATED, "Failed to add custom field during setup")
+
+    def test_delete_custom_field_success(self):
```

Comment:
Lets add all types of custom fields

INT BOOLEAN FLOAT VARCHAR(255) etc and then verify them by listing all of them and checking and then lets test deleting them as a testcase one by one, and then list all the custom fields one by one and verify the deleted custom field is not present

Add 2 or 3 custom fields and delete one of them and verify the remaining 2 custom fields are present

Try giving a wrong URL  like

{self.BASE_URL}/custom_fields/{'test_field'}/blah/blah/blah?customer_guid={self.valid_customer_guid}" and test that this URL should not work. It should not work by default, as FastAPI takes care of all these routing but you can test once as a testcase

Reply:
modified sir